packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -140,7 +113,7 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
             // cases, but since this is a contract we'd end up bumping the nonce twice.
             Lib_ExecutionManagerWrapper.ovmINCREMENTNONCE();
 
-            return decodedTx.to.call(decodedTx.data);
+            return transaction.to.call{gas: transaction.gasLimit}(transaction.data);
```

Comment:
I think that modifying how gas is handled should probably be treated in a separate PR for separation of concerns, and because it impacts the fee logic other folks are working on.  This syntax [should work though](https://github.com/ethereum-optimism/solidity/blob/bfa0224dd19a3a5101242f281ec81d0f02b9ee3a/libsolidity/codegen/CompilerContext.cpp#L146) once we do want to add it.

Reply:
RIP - noticing now that we actually accidentally *removed* the gasLimit logic in this PR from yesterday: https://github.com/ethereum-optimism/optimism/pull/546/files. So might be best to add it back in...

Reply:
Ahhhh snap, you are right.  Should've caught that in 546.  In that case let's do it here, but cc @K-Ho who I think has some of the best insight into this right now.

Reply:
NVM sorry I missed *again* that we were passing `gasleft()` so the first version here was the correct one.